portfolio/src/main/webapp/script-home.js

```diff
@@ -33,38 +33,69 @@ window.onload = function() { main(); }
  * function main() initializes the slideshows and the interactive elements on the website.
  */
 function main() {
-    const /** ?HTMLCollection */ slideShowGallery =
-        new SlideShow(document.getElementsByClassName('gallery-slides'));
-    const /** ?HTMLCollection */ slideShowBlog  =
-        new SlideShow(document.getElementsByClassName('blog-slides'));
+    initializeSlideshows();
+    populateComments();
+}
 
-    slideShowBlog.setToAutomaticallyChangeSlides();
+/** 
+ * function initializeSlideshows() initializes the blog slideshow and the gallery slideshow.
+ */
+function initializeSlideshows() {
+  const /** ?HTMLCollection */ slideshowGallery =
+      new SlideShow(document.getElementsByClassName('gallery-slides'));
+  const /** ?HTMLCollection */ slideshowBlog  =
+      new SlideShow(document.getElementsByClassName('blog-slides'));
+  slideshowBlog.setToAutomaticallyChangeSlides();
 
-    document.getElementById('switch-slides-left').onclick =
+  document.getElementById('switch-slides-left').onclick =
         function adjustBackOne() {
           slideShowGallery.adjustSlideManual(ADJUST_BACK); 
-    }
-    document.getElementById('switch-slides-right').onclick =
+        };
+  document.getElementById('switch-slides-right').onclick =
         function adjustForwardOne() {
           slideShowGallery.adjustSlideManual(ADJUST_FORWARD); 
-    }
-    // Fetches the pre-made comment list from the servlet and makes 
-    // a list for each comment.
-    fetch('/comments').then(response => response.json()).then((comments) => {
-      const /** ?HTMLCollection */commentContainer =
-            document.getElementById('comments-container');
-      comments.forEach(function(comment) {
-          commentContainer.appendChild(createListElement(comment));
-      });
-    });
+        };
 }
 
-/*
- * @return the <li> element containing text passed in.
- * @param {string} text is what is put into the <li>
+/** 
+ * function populateComments() populates the comment board on the webpage.
  */
-function createListElement(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
-  return liElement;
+function populateComments() {
+  fetch('/comments').then(response => response.json()).then(
+        (comments) => {
+          const /** ?HTMLCollection */commentContainer =
+              document.getElementById('comments-container');
+
+          comments.forEach(function(comment) {
+            let /** string */ stringOfName;
+            if (comment.name==null || comment.name=='') {
```

Comment:
comment.name<space>==<space>null

And everywhere else

Reply:
Also, you want:

comment.name ==(2 or 3 equals is fine) null || comment.name ===(3 equals) "" || comment.name === undefined

=== compares type + content 
== only compares type